### PR TITLE
Checked select all checkboxes if all options are checked

### LINKED ISSUE #18858: Select all not being selected
Hi select all in the below screenshot, either both the dropdown value along with select all should be selected or none of the options has to be selected by default.
Screenshot from 2020-04-28 12-17-39

### PR DESCRIPTION
**Pull request by tegbessou**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Checked select all checkboxes if all options are checked<br>Add in the `buildView` method of `MaterialChoiceTableType` a variable to check 'Select all' checkbox and use this variable in `bootstrap_4_layout.html.twig` and `prestashop_ui_kit_base.html.twig` to apply this change to all forms which used the `MaterialChoiceTableType`
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #18858.
| How to test?      | Go to Catalog > Categories and update a category if on field 'Group access' all value are checked, so the 'Select All' checkbox must be checked. If all values are not checked, check all values and then come back to the edit page and the 'Select All' checkbox must be checked.
| Possible impacts? | Select all check box also be checked if all choices are checked on:<br>- International > Localization > Geolocation<br>- Catalog > Categories (try to update a root category)<br>- Customers > Addresses > Set required fields for this section<br>- Customers > Customers (edit a customer and look GroupAccess field)<br>- Customers > Customers > Set required fields for this section

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26215)

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php [hunk 1]
```diff
@@ -54,6 +56,14 @@ public function getParent()
         return ChoiceType::class;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options): void
```
Comment (Progi1984) [inline src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php:62]: Could you add an unit test, please ?

File: src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig [hunk 2]
```diff
@@ -586,7 +586,13 @@
             <th class="checkbox">
               <div class="md-checkbox">
                 <label>
-                  <input type="checkbox" class="js-choice-table-select-all">
+                  <input
+                    type="checkbox"
+                    class="js-choice-table-select-all"
+                    {% if isCheckSelectAll %}
+                      checked
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig:593]: @NeOMakinG `checked` or `checked="checked"` ?

**Review by Progi1984 — COMMENTED** [review]
@tegbessou Thank you for your contribution. I have brought one small feedback  :)  and a question for @NeOMakinG

File: src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php (hunk 1 above)
Comment (tegbessou) [reply src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php:62]: Yes sure, i will do this!! Thank for your review!

File: src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig (hunk 2 above)
Comment (NeOMakinG) [reply src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig:593]: `checked` is fine, it just need to be defined

Comment (tegbessou) [reply src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig:593]: So i can resolve this ?

File: src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php (hunk 1 above)
Comment (tegbessou) [reply src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php:62]: I add it! This is good for you @Progi1984 ?

File: tests/Unit/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Unit\PrestaShopBundle\Form\Admin\Type\Material;
```
Comment (Progi1984) [inline tests/Unit/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableTypeTest.php]:
```suggestion
 */

declare(strict_types=1);

namespace Tests\Unit\PrestaShopBundle\Form\Admin\Type\Material;
```

File: src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php (hunk 1 above)
Comment (Progi1984) [reply src/PrestaShopBundle/Form/Admin/Type/Material/MaterialChoiceTableType.php:62]: Perfect. A small comment in test and it's approved.

**Review by Progi1984 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]
LGTM.
Maybe @NeOMakinG you can also improve the UX by adding a little bit of js to automatically check or uncheck depending of what you checked :sweat_smile: :
image
image

**Review by matks — APPROVED** [review]

**Comment by marwachelly** [comment]
Hello,
PR is tested and ok in the following cases:
- International > Localization > Geolocation
- Catalog > Categories (try to update a root category)
- Customers > Addresses > Set required fields for this section
- Customers > Customers (edit a customer and look GroupAccess field)
- Customers > Customers > Set required fields for this section

https://www.awesomescreenshot.com/video/5774911?key=77183165a07a7a474df7ffe1f82bbb82

Many thanks,

**Comment by Progi1984** [comment]
Thanks @tegbessou & @marwachelly

**Pull request merged**